### PR TITLE
Implement Balancer v2 Origin[sc-2666]

### PR DESCRIPTION
**Pull request by deep-quality-dev**
### description of pull request:

Implemented Balancer v2 origin & update the config_next.hcl.

Here is the example of origins which integrates with Blockchain:
`type` indicates the origin type, `default` as the label of `contracts` block indicates the Ethereum Client like `ethereum`, `arbitrum`, `optimism`.
Any origins may have number of `contracts` blocks if they integrates multiple blockchains for fetching prices.
```
  origin "balancerV2" {
    type = "balancerV2"
    contracts "ethereum" {
      addresses = {
        "WETH/GNO" = "0xF4C0DD9B82DA36C07605df83c8a416F11724d88b"
        "RETH/WETH" = "0x1E19CF2D73a72Ef1332C882F20534B6519Be0276"
        "STETH/WETH" = "0x32296969ef14eb0c6d29669c550d4a0449130230"
        "WETH/YFI" = "0x186084ff790c65088ba694df11758fae4943ee9e"
      }
      references = {
        "RETH/WETH" = "0xae78736Cd615f374D3085123A210448E74Fc6393"
      }
    }
  }
```

### CODE REVIEW

**Comment by deep-quality-dev** [comment]
Please don't start review while PR is under [WIP].

File: pkg/datapoint/origin/balancer_v2.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+//go:embed balancerv2_abi.json
+var balancerV2PoolABI []byte
+
+const BalancerV2LoggerTag = "BALANCERV2_ORIGIN"
+
+type BalancerV2Options struct {
+	Client             rpc.RPC
+	ContractAddresses  ContractAddresses
+	ReferenceAddresses ContractAddresses
```
Comment (MDobak) [inline pkg/datapoint/origin/balancer_v2.go:31]: What's the difference between `ContractAddresses` and `ReferenceAddresses`?

File: pkg/datapoint/origin/balancer_v2.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	"github.com/chronicleprotocol/oracle-suite/pkg/log/null"
+	"github.com/chronicleprotocol/oracle-suite/pkg/util/bn"
+)
+
+//go:embed balancerv2_abi.json
+var balancerV2PoolABI []byte
+
+const BalancerV2LoggerTag = "BALANCERV2_ORIGIN"
+
+type BalancerV2Options struct {
```
Comment (MDobak) [inline pkg/datapoint/origin/balancer_v2.go:28]: A small thing, but everywhere else in the code we call this type of structure `Config`. The word `Option` suggests the use of `Options Pattern`.

File: pkg/datapoint/origin/balancer_v2.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+		}
+		points[pair] = datapoint.Point{
+			Value: tick,
+			Time:  time.Now(),
+		}
+	}
+
+	if len(pairs) == 1 && points[pairs[0]].Error != nil {
+		return points, points[pairs[0]].Error
+	}
```
Comment (MDobak) [inline pkg/datapoint/origin/balancer_v2.go:202]: I think this check is not necessary. The error argument is reserved for situations where there is a general problem that prevents you from getting any data from the origin. If you can attach an error to the data point, that's enough.

File: pkg/datapoint/origin/balancer_v2.go (hunk 2 above)
Comment (deep-quality-dev) [reply pkg/datapoint/origin/balancer_v2.go:28]: Perfect, I will rename it with `Config` structure for the other origins, I just noticed jq origin has `TickGenericJQOptions` structure, will create another PR for jq origin.

File: pkg/datapoint/origin/balancer_v2.go (hunk 3 above)
Comment (deep-quality-dev) [reply pkg/datapoint/origin/balancer_v2.go:202]: Sure, actually they are unnecessary code. Good point

**Review by teghnet — APPROVED** [review]

**Pull request merged**